docs/specs/clients/notify/client-sdk-api.md

```diff
@@ -35,38 +37,54 @@ abstract class Client {
     account: string,
   }): Promise<Record<string, NotifySubscription>>;
 
-  // get all messages for a subscription
-  public abstract getMessageHistory(params: {
+  // get all notifications for a subscription
+  public abstract getNotificationHistory(params: {
     topic: string,
-  }): Promise<Record<number, NotifyMessageRecord>>
+  }): Promise<Record<number, NotifyNotificationRecord>>
 
   // delete active subscription
   public abstract deleteSubscription(params: {
     topic: string,
   }): Promise<void>;
   
-  // delete notify message
-  public abstract deleteNotifyMessage(params: {
+  // delete notification
+  public abstract deleteNotification(params: {
     id: number,
   }): Promise<void>;
   
   // decrypt notify subscription message
-  public abstract decryptMessage(params: {
+  public abstract decryptNotification(params: {
     topic: string,
     encryptedMessage: string,
-  }): Promise<NotifyMessage>;
+  }): Promise<NotifyNotification>;
+  
+  // Generates and returns SIWE message along with
+  // Cacao Payload with identity key
+  public abstract prepareRegistration(params: {
+    account: string;
+    domain: string,
+    // Defaulted to false if not passed
+    allApps?: boolean,
+  }): {
+    registerParams: NotifyRegistrationParams,
+    message: string
+  };
   
   // "Logs in" this client to notifications for the specified account. This involves:
-  // - Generating, signing, saving to local stroage, and registering an Identity Key, if one didn't already exist in local storage
+  // - Verifies signature by reconstructing formatted message using provided payload
+  // - Expected payload comes from prepareRegistration
+  // - Once verified, registers the identity key on the keyserver
   // - Calls `wc_notifyWatchSubscriptions`, watches for future subscription updates, and subscribes to all notification topics
-  // Returns the Identity Key public key. Method should throw 'signatureRejected' if any errors comes from onSign promise. 
+  // Returns the Identity Key public key. 
   public abstract register(params: {
-    account: string;
-    private?: boolean;
-    domain: string,
-    isLimited?: boolean,
-    onSign: (message: string) => Promise<Cacao.Signature>
+    registerParams: NotifyRegistrationParams,
```

Comment:
We're planning to take account from Cacao.Payload? May be we can pass it as method param?

Reply:
Yes in my implementation I'm taking it from payload